luarules/gadgets/unit_stockpile_limit.lua

```diff
@@ -28,14 +28,19 @@ if gadgetHandler:IsSyncedCode() then
 	local GiveOrderToUnit	= Spring.GiveOrderToUnit
 
 	local canStockpile = {}
+	Spring.Log(gadget:GetInfo().name, LOG.WARNING, "Check all stockpiles "..#UnitDefs)
 	for udid, ud in pairs(UnitDefs) do
 		if ud.canStockpile then
 			canStockpile[udid] = true
 		end
-		if ud.customParams.stockpileLimit then
-			isStockpilingUnit[udid] = tonumber(ud.customParams.stockpileLimit)
-		elseif ud.customParams.stockpilelimit then
-			isStockpilingUnit[udid] = tonumber(ud.customParams.stockpilelimit)
+		if ud.weapons then
+			for i = 1, #ud.weapons do
+				local weaponDef = WeaponDefs[ud.weapons[i].weaponDef]
```

Comment:
Isn't `weaponDef` already the weapon def?
```suggestion
			for _, weaponDef in ipairs(ud.weapons) do
```

Reply:
IIRC it's just the ID.

Reply:
`weaponDef.weaponDef` is the weapon def ID. From the wiki example, it looks like `weapons` table is an array where each value is the weapon def.
https://springrts.com/wiki/Lua_UnitDefs

Reply:
Yeah, that's why I added (a lot) of logging, because I couldn't work out the steps I had to go through to get the stockpilelimit. What's there definitely works and I can't see any way, documented or otherwise, to get the value out.